### ISSUE TITLE
The problem that adm-zip reads zip files takes up a lot of memory！

### ISSUE THREAD

**Issue by wangmings** [cthackers/adm-zip#417]
### The problem that adm-zip reads zip files takes up a lot of memory

**Comment by congzhangzh** [comment]
I face the problem too, si that possible to make the get entities as a lazy iterator?

**Comment by congzhangzh** [comment]
It seems it already support it(do not use getEntities is a key part)?

https://github.com/cthackers/adm-zip/blob/bebbabf600a6ae5d64ee31cbc89a1274bd1d8c1f/zipFile.js#L140

https://github.com/cthackers/adm-zip/blob/bebbabf600a6ae5d64ee31cbc89a1274bd1d8c1f/zipFile.js#L148

https://github.com/cthackers/adm-zip/blob/bebbabf600a6ae5d64ee31cbc89a1274bd1d8c1f/zipFile.js#L25

@wangmings can you have a try?

**Comment by wangmings** [comment]
### I tried to replace getEntries with getEntryCount and it couldn't be executed. I didn't study the specific source code! The process of decompressing zip files is a process from small to large! In this decompression process, a lot of operations are required, which will inevitably take up a lot of memory resources! I think this question should be unsolved!

**Comment by melroy89** [comment]
I think there is also a memory leak present in `adm-zip` or memory fragmentation under Linux. Causing also memory to increase every time I unzip the same file.

**Comment by mbehr1** [comment]
I'm trying to add support for Blob instead of Buffer so that only the needed data could be put into memory with .slice(...).
This should allow opening huge zip files (e.g. 50GB) as well (as long as the single files to extract are still fitting into an ArrayBuffer.)

**Comment by lindstrm** [comment]
> I'm trying to add support for Blob instead of Buffer so that only the needed data could be put into memory with .slice(...). This should allow opening huge zip files (e.g. 50GB) as well (as long as the single files to extract are still fitting into an ArrayBuffer.)

How did this go, anything merged?

**Comment by darkostanimirovic** [comment]
Support for streaming would be appreciated, I'm trying to extract files > 2 GiB.

**Comment by melroy89** [comment]
I moved to Go.. Problem solved 🥳 . https://gitlab.melroy.org/melroy/gitlab-artifact-deployer-go/-/blob/main/server.go?ref_type=heads#L181

**Comment by MorganVonBrylan** [comment]
> > > I think there is also a memory leak present in `adm-zip` or memory fragmentation under Linux. Causing memory to increase every time I unzip the same file. And not giving the memory allocation back to the system. 😢
> > 
> > 
> > Can confirm the same thing. I used adm-zip to make daily backups of my sqlite database, using one zip file per month. But as the day go, the memory usage quickly increases by hundreds of megabytes, even though the unzipped contents only weigh 20 to 40 MB (a lot less at the beginning of the month when a new zip file is made).
> 
> Good to know it's not just me!
> 
> So my solution was simple: I got sick of Javascript and rewrote my tool in Golang. Go is not only faster it's has no memory leaks now and the overall memory concuption is also down.

Yeah I saw that. It's a shame to have to learn an entirely new language just to solve an issue that shouldn't exist to begin with, though.

**Comment by melroy89** [comment]
True, but I glad I move to Go for this. I still like typescript, but I wouldn't have used that language for writing my actual tools or basic services. I went with the hype too much maybe.

It has a learning curve for sure. But you won't be disappointed.
